flutter/lib/src/integrations/native_app_start_integration.dart

```diff
@@ -21,18 +26,98 @@ class NativeAppStartIntegration extends Integration<SentryFlutterOptions> {
         options.logger(SentryLevel.debug,
             'Scheduler binding is null. Can\'t auto detect app start time.');
       } else {
-        schedulerBinding.addPostFrameCallback((timeStamp) {
+        schedulerBinding.addPostFrameCallback((timeStamp) async {
           // ignore: invalid_use_of_internal_member
-          _native.appStartEnd = options.clock();
+          final appStartEnd = options.clock();
+          _native.appStartEnd = appStartEnd;
+
+          if (!_native.didFetchAppStart) {
+            final nativeAppStart = await _native.fetchNativeAppStart();
+            final measurement = nativeAppStart?.toMeasurement(appStartEnd!);
+
+            // We filter out app start more than 60s.
+            // This could be due to many different reasons.
+            // If you do the manual init and init the SDK too late and it does not
+            // compute the app start end in the very first Screen.
+            // If the process starts but the App isn't in the foreground.
+            // If the system forked the process earlier to accelerate the app start.
+            // And some unknown reasons that could not be reproduced.
+            // We've seen app starts with hours, days and even months.
+            if (nativeAppStart == null ||
+                measurement == null ||
+                measurement.value >= 60000) {
+              _appStartTracker?.setAppStartInfo(null);
+              return;
+            }
+
+            final appStartInfo = AppStartInfo(
+              DateTime.fromMillisecondsSinceEpoch(
+                  nativeAppStart.appStartTime.toInt()),
+              appStartEnd,
+              measurement,
+            );
+
+            _appStartTracker?.setAppStartInfo(appStartInfo);
+          } else {
+            _appStartTracker?.setAppStartInfo(null);
+          }
         });
       }
     }
 
-    options.addEventProcessor(NativeAppStartEventProcessor(_native));
+    options.addEventProcessor(NativeAppStartEventProcessor(appStartTracker: _appStartTracker));
 
     options.sdk.addIntegration('nativeAppStartIntegration');
   }
 }
 
 /// Used to provide scheduler binding at call time.
 typedef SchedulerBindingProvider = SchedulerBinding? Function();
+
+@internal
+class AppStartInfo {
+  final DateTime start;
+  final DateTime end;
+  final SentryMeasurement measurement;
+
+  AppStartInfo(this.start, this.end, this.measurement);
+}
+
+abstract class IAppStartTracker {
+  AppStartInfo? get appStartInfo;
+
+  void setAppStartInfo(AppStartInfo? appStartInfo);
+
+  void onAppStartComplete(Function(AppStartInfo?) callback);
+}
+
+@internal
+class AppStartTracker extends IAppStartTracker {
```

Comment:
`l`: Why did you add the interface `IAppStartTracker`? I'm a fan of only creating interfaces if you will have multiple implementations. Some people refer to this as interface overuse. I call it YAGNI (You Aren't Gonna Need It). Right now it looks like you could ditch the interface to simplify things.